src/wallet/rpcwallet.cpp

```diff
@@ -4438,7 +4438,9 @@ static RPCHelpMan upgradewallet()
         {
             {"version", RPCArg::Type::NUM, /* default */ strprintf("%d", FEATURE_LATEST), "The version number to upgrade to. Default is the latest wallet version"}
         },
-        RPCResults{},
+        RPCResult{
+            RPCResult::Type::STR, "Error", "error description during upgrading wallet"
```

Comment:
Would be good to return an object instead. See

```
doc/developer-notes.md:- Try to make the RPC response a JSON object.

Reply:
should it return an object even it's only one value?

Reply:
RPC Result changed from `str` to `object`